source/adapters/hip/queue.hpp

```diff
@@ -261,3 +267,26 @@ struct ur_queue_handle_t_ {
 
   bool backendHasOwnership() const noexcept { return HasOwnership; }
 };
+
+// RAII object to make hQueue stream getter methods all return the same stream
+// within the lifetime of this object.
+//
+// This is useful for urEnqueueNativeCommandExp where we want guarantees that
+// the user submitted native calls will be dispatched to a known stream, which
+// must be "got" within the user submitted function.
+//
+// TODO: Add a test that this scoping works
```

Comment:
Nit:

To make sure that this TODO doesn't get overlooked it might make sense to also copy it to the cuda adapter, which also uses the same scoping mechanism.

Reply:
Good idea thanks Martin, will add a new test tomorrow